tests/examples/wibox/layout/template.lua

```diff
@@ -22,15 +22,20 @@ local function generic_widget(text, col)
                 end,
                 widget = wibox.widget.base.make_widget
             },
-            text and {
+            {
+                id     = "text",
                 align  = "center",
                 valign = "center",
-                text   = text,
+                text   = text or "foobar",
                 widget = wibox.widget.textbox
             } or nil,
             widget = wibox.layout.stack
         },
         margins = 5,
+        set_text = function(self, text2)
+            self:get_children_by_id("text")[1]:set_text(text2)
+        end,
+        is_widget = true,
```

Comment:
Huh? This `is_widget` seems fishy to me, why is it needed?
Also, you might want to rebase this on master. Half of the commit is already applied.